### PR TITLE
FW Dump Me HLD

### PR DESCRIPTION
**Pull request by shlomibitton**
Signed-off-by: Shlomi Bitton <shlomibi@mellanox.com>

### CODE REVIEW

File: doc/fw_dump_me/FW-Dump-Me-in-SONiC.md [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+2b199b2309f9        docker-fw-dump-me:latest             "/usr/bin/supervisord"   17 hours ago        Up 11 hours                             docker-fw-dump-me
+```
+
+* SDK Unix socket needs to be mapped to container (for CLI support).
+* */var/log/mellanox/* mounted inside container (used for writing dump files)
+* */var/run/fw_dump_me/* mounted inside container (used for Unix domain socket)
+
+## 3.3 FW Dump Me in SONiC overview 
+
+![FW Dump Me in SONiC overview](/doc/fw_dump_me/overview.svg)
```
Comment (keboliu) [inline doc/fw_dump_me/FW-Dump-Me-in-SONiC.md:110]: regarding the chart, Does the fw_dumpd need to talk to sx_sdk inside syncd or talk to the sx_core on the host?

File: doc/fw_dump_me/FW-Dump-Me-in-SONiC.md [hunk 2]
```diff
@@ -0,0 +1,282 @@
...
+## 3.6 FW Dump Me provided data
+
+- The generating cause & time (the cause of the “FW dump me now trap” or the reason the high level decided to take the dump).
+- FW trace, I.E the FW configuration done to the HW from the beginning of time (boot). 
+- GDB core files of all irisc’s.
+- As much of the CR space according to priority defined in the ADB file.
+
+## 3.7 FW Dump Me output file
+
+The output file will be generated in "/var/log/mellanox/" if triggered by FW or SDK cause.
```
Comment (keboliu) [inline doc/fw_dump_me/FW-Dump-Me-in-SONiC.md:178]: I suggest to create a subfolder, we already have a subfolder 'sniffer' for SDK sniffer files.

File: doc/fw_dump_me/FW-Dump-Me-in-SONiC.md (hunk 1 above)
Comment (shlomibitton) [reply doc/fw_dump_me/FW-Dump-Me-in-SONiC.md:110]: I am not sure but I think both.
the sx_sdk on syncd is for registering to the trap group of SDK health events.
the sx_core is to "listen" to sdk health events and trap them to trigger the dump taking.

File: doc/fw_dump_me/FW-Dump-Me-in-SONiC.md (hunk 2 above)
Comment (shlomibitton) [reply doc/fw_dump_me/FW-Dump-Me-in-SONiC.md:178]: sure, changed it to "/var/log/mellanox/fw_dump"

File: doc/fw_dump_me/FW-Dump-Me-in-SONiC.md (hunk 1 above)
Comment (shlomibitton) [reply doc/fw_dump_me/FW-Dump-Me-in-SONiC.md:111]: understood

File: doc/fw_dump_me/FW-Dump-Me-in-SONiC.md [hunk 3]
```diff
@@ -0,0 +1,282 @@
...
+
+The new dump file name will be: <device_id>/<module_name>_<time_stamp>
+
+## 3.8 CLI
+
+Since the FW Dump Me daemon is running on a seperate container, CLI will be provided by an additional thread communicating with the host with a socket. The thread is running on the docker, recieve and process requests from the host to trigger dump generating.
+
+The command to create a new dump:
+```
+admin@sonic:~$ fwdumpme <desired_path>
```
[the remark this answers is not in the document]
Comment (shlomibitton) [reply doc/fw_dump_me/FW-Dump-Me-in-SONiC.md]: examples added

**Pull request closed without merging**